test/e2e/scheduler_predicates.go

```diff
@@ -900,16 +886,37 @@ func getRequestedCPU(pod v1.Pod) int64 {
 	return result
 }
 
-func waitForScheduler() {
-	// Wait a bit to allow scheduler to do its thing
-	// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
-	framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
-	time.Sleep(10 * time.Second)
+func waitForScheduler(f *framework.Framework, PodName string, expectSuccess bool) {
+	framework.Logf("Waiting for scheduler...")
+	action := func() error { // No action before waiting for Scheduler.
+		return nil
```

Comment:
Side note 3 (apologies): cached events, if they do occur, could cause false positives for the scheduleSuccess event predicate if pod names are not unique within the test run namespace. We should make sure pod names are not re-used for the lifetime of the namespace to avoid that problem.